packages/client/src/provider/in-memory-provider/in-memory-provider.ts

```diff
@@ -0,0 +1,186 @@
+import {
+  EvaluationContext,
+  FlagNotFoundError,
+  FlagValueType,
+  GeneralError,
+  JsonValue,
+  Logger,
+  OpenFeatureError,
+  OpenFeatureEventEmitter,
+  ProviderEvents,
+  ResolutionDetails,
+  StandardResolutionReasons,
+  TypeMismatchError,
+  ProviderStatus,
+  Flag,
+  FlagConfiguration,
+  VariantFoundError,
+} from '@openfeature/shared';
+import { Provider } from '../provider';
+
+/**
+ * A simple OpenFeature provider intended for demos and as a test stub.
+ */
+export class InMemoryProvider implements Provider {
+  public readonly events = new OpenFeatureEventEmitter();
+  public readonly runsOn = 'client';
+  status: ProviderStatus = ProviderStatus.NOT_READY;
+  readonly metadata = {
+    name: 'in-memory',
+  } as const;
+  private _flagConfiguration: FlagConfiguration;
+  private _flagEvalCache: Record<string, ResolutionDetails<number | string | boolean | JsonValue>> = {};
+
+  constructor(flagConfiguration: FlagConfiguration = {}) {
+    this._flagConfiguration = { ...flagConfiguration };
+  }
+
+  async initialize(context?: EvaluationContext | undefined): Promise<void> {
+    this.status = ProviderStatus.READY;
+    try {
+
+      for (const key in this._flagConfiguration) {
+        if (this._flagConfiguration[key].disabled) {
+          continue;
+        }
+
+        this._flagEvalCache[key] = this.resolveFlagWithReason(key, context);
+
+      }
+
+      this.events.emit(ProviderEvents.Ready);
+    } catch (error) {
+      this.status = ProviderStatus.ERROR;
+      this.events.emit(ProviderEvents.Error);
+      throw error;
+    }
+  }
+
+  /**
+   * Overwrites the configured flags.
+   * @param { FlagConfiguration } flagConfiguration new flag configuration
+   */
+  putConfiguration(flagConfiguration: FlagConfiguration) {
+    const flagsChanged = Object.entries(flagConfiguration)
+      .filter(([key, value]) => this._flagConfiguration[key] !== value)
+      .map(([key]) => key);
+
+    this.status = ProviderStatus.STALE;
+    this.events.emit(ProviderEvents.Stale);
```

Comment:
This behavior doesn't match what was done in the server SDK.

https://github.com/open-feature/js-sdk/blob/main/packages/server/src/provider/in-memory-provider/in-memory-provider.ts

Reply:
The idea here was to mimic the state of a "real" client side providers, since it evaluates all flags at once.
I think the server version of the in memory provider can also set it's state to stale and emit the stale event when it begins to process the new config file, but since it's a synchronous process it wouldn't be much valuable. It would also need to emit Ready and set state when it finishes. 

In terms of behavior that's ok, but I don't know how valuable this is in practice.